### PR TITLE
Bug/various fixes for tests and CI

### PR DESCRIPTION
**Pull request by abelikt**
## Proposed changes

A bunch of cherry-picks to fix various issues in the official CI chain.
- detection of rolldice debs
- Add junit report handlers to system-test python environment
Also pick un-merged docu and typo-fixes.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: tests/requirements.txt [hunk 1]
```diff
@@ -2,4 +2,6 @@ pysys == 1.6.1
 requests == 2.25.1
 toml == 0.10.2
 retry == 0.9.2
+junitparser == 2.2.0
+junit2html == 30.0.6
```
Comment (albinsuresh) [inline tests/requirements.txt:6]: These changes are unrelated to this fix, right? Accidental commit from your other work trying to generate Junit style test reporting for pysys?

**Review by albinsuresh — APPROVED** [review]

**Review by PradeepKiruvale — APPROVED** [review]
LGTM

File: tests/PySys/software_management_end_to_end/dummy_plugin_configuration/Info.md [hunk 2]
```diff
@@ -19,12 +19,12 @@ When executed via the software management agent, this path will be used:
...
-The current trick is just to use a static configuration to emulate a funcional
-plugin that lets us install and deinstall fruits.
+The current trick is just to use a static configuration to emulate a functional
+plugin that lets us install and de-install fruits.
 Right now, the apt plugin only cares about the returned text on the list command.
 So we just insert a static list and base tests on that.
 
-Note that even when a package is deinstalled it will still appear in the list
+Note that even when a package is de-installed it will still appear in the list
 as the response is static.
```
Comment (didier-wenzek) [inline tests/PySys/software_management_end_to_end/dummy_plugin_configuration/Info.md:28]: I see that as an important issue to be fixed in the dummy plugin.

* A plugin implemented for testing purpose *must* follow the requirements.
* Using a static file to mimic the repository is fine but not to manage what is installed or not.

File: tests/requirements.txt (hunk 1 above)
Comment (abelikt) [reply tests/requirements.txt:6]: That's a separate fix. We need both of the packages to analyse records.

File: tests/PySys/software_management_end_to_end/dummy_plugin_configuration/Info.md (hunk 2 above)
Comment (abelikt) [reply tests/PySys/software_management_end_to_end/dummy_plugin_configuration/Info.md:28]: The dummy plugin that we currently have responds with a static response loaded from a text-config file. Currently, it has no other state. We could update the config file to change the response (or rewrite the plugin).

File: tests/PySys/software_management_end_to_end/sm_fake_install_remove/run.py [hunk 3]
```diff
@@ -30,7 +30,8 @@ class PySysTest(SoftwareManagement):
     def setup(self):
 
         if self.fakeplugin != "fakeplugin":
-            self.skipTest("Testing the apt plugin is not supported on this platform")
+            self.skipTest("Testing the fake plugin is not supported on this platform"+\
```
[the remark this answers is not in the document]
Comment (abelikt) [reply tests/PySys/software_management_end_to_end/sm_fake_install_remove/run.py]: Thanks for the hint that message is a bit confusing here. I updated it.
